### PR TITLE
Add stripe checkout

### PR DESCRIPTION
**Pull request by czakuou**

### CODE REVIEW

File: src/migrations/env.py [hunk 1]
```diff
@@ -1,17 +1,20 @@
 import asyncio
```
Comment (jzyworonek) [inline src/migrations/env.py:1]:
```suggestion
from __future__ import annotations

import asyncio
```

File: src/migrations/env.py [hunk 2]
```diff
@@ -40,9 +43,9 @@ def run_migrations_offline() -> None:
         context.run_migrations()
 
 
-def do_run_migrations(connection: Connection) -> None:
+def do_run_migrations(connection: "Connection") -> None:
```
Comment (jzyworonek) [inline src/migrations/env.py:46]:
```suggestion
def do_run_migrations(connection: Connection) -> None:
```

File: src/migrations/env.py [hunk 3]
```diff
@@ -40,9 +43,9 @@ def run_migrations_offline() -> None:
...
     def process_revision_directives(
-        context: MigrationContext, _revision: tuple[str], directives: list[MigrationScript]
+        context: "MigrationContext", _revision: tuple[str], directives: list["MigrationScript"]
```
Comment (jzyworonek) [inline src/migrations/env.py:48]:
```suggestion
        context: MigrationContext, _revision: tuple[str], directives: list[MigrationScript]
```

File: tests/test_create_checkout.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+from typing import TYPE_CHECKING
```
Comment (jzyworonek) [inline tests/test_create_checkout.py:1]:
```suggestion
from __future__ import annotations

from typing import TYPE_CHECKING
```

File: tests/test_create_checkout.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+from unittest.mock import Mock, patch
+
+import stripe
+
+if TYPE_CHECKING:
+    from fastapi.testclient import TestClient
+
+
+@patch("stripe.checkout.Session.create")
+def test_create_checkout_should_return_url(mock_stripe: Mock, client: "TestClient") -> None:
```
Comment (jzyworonek) [inline tests/test_create_checkout.py:11]:
```suggestion
def test_create_checkout_should_return_url(mock_stripe: Mock, client: TestClient) -> None:
```

File: tests/test_create_checkout.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            "payment_method_types": ["card"],
+        },
+    )
+
+    assert response.status_code == 201
+    assert response.json() == {"url": "https://example.com/checkout"}
+
+
+@patch("stripe.checkout.Session.create")
+def test_test_checkout_stripe_error_should_rise_checkout_error(mock_stripe: Mock, client: "TestClient") -> None:
```
Comment (jzyworonek) [inline tests/test_create_checkout.py:27]:
```suggestion
def test_create_checkout_raises_checkout_error(mock_stripe: Mock, client: TestClient) -> None:
```

File: pyproject.toml [hunk 7]
```diff
@@ -17,6 +17,9 @@ structlog = "^24.1.0"
 alembic = "^1.13.1"
 pre-commit = "^3.7.0"
 sqlalchemy = {version = "^2.0.29", extras = ["asyncio", "postgresql-psycopgbinary"]}
+stripe = "^9.3.0"
+pydantic-extra-types = "^2.7.0"
+pycountry = "^23.12.11"
```
Comment (jzyworonek) [inline pyproject.toml:22]: where is pycountry being used?

**Review by jzyworonek — COMMENTED** [review]
I'm not feeling the structure but maybe one day 🙈

File: src/features/create_checkout.py [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+        self._success_url = success_url
+
+    def create_checkout(self, data: CreateCheckoutRequest) -> dict[Any, Any]:
+        self._log.info("Creating a new checkout session.")
+        try:
+            session = stripe.checkout.Session.create(
+                payment_method_types=data.payment_method_types,  # type: ignore[arg-type]
+                line_items=[
+                    {
+                        "price_data": {
```
Comment (matkubaszek) [inline src/features/create_checkout.py:52]: I will try move data structure out of try. Before session create

File: src/features/create_checkout.py [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+
+
+CheckoutServiceDependency = Annotated[CheckoutService, Depends(get_stripe_connector)]
+
+
+@router.post(
+    "/payments",
+    response_model=CreateCheckoutResponse,
+    summary="Create a new checkout session.",
+    status_code=status.HTTP_201_CREATED,
```
Comment (matkubaszek) [inline src/features/create_checkout.py:84]: This is very "unclear" in IT community. If we call other service better will be https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202

File: tests/test_create_checkout.py (hunk 5 above)
Comment (matkubaszek) [inline tests/test_create_checkout.py:11]: test_should_create_new_checkout_session <- we test behave, not what will be returned

File: src/features/create_checkout.py [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+@cache
+def get_stripe_connector(config: StripeConfigDependency, logger: LoggerDependency) -> CheckoutService:
+    return CheckoutService(stripe_api_key=config.api_key, logger=logger, success_url=config.success_url)
+
+
+CheckoutServiceDependency = Annotated[CheckoutService, Depends(get_stripe_connector)]
+
+
+@router.post(
+    "/payments",
```
Comment (matkubaszek) [inline src/features/create_checkout.py:81]: api/payment 
? this doesn't say much about process
api/payment/checkout will be much better

File: src/features/create_checkout.py [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+                            "unit_amount": int(data.amount.quantize(Decimal("1.00")) * 100),
+                        },
+                        "quantity": 1,
+                    },
+                ],
+                mode="payment",
+                success_url=self._success_url,
+            )
+        except stripe.StripeError as e:
+            self._log.error("An error occurred while creating the checkout.", exc_info=e)
```
Comment (matkubaszek) [inline src/features/create_checkout.py:66]: And passed session id to identify

File: pyproject.toml (hunk 7 above)
Comment (czakuou) [reply pyproject.toml:22]: by `Currency`

**Pull request merged**